src/NLog/LogEventInfo.cs

```diff
@@ -281,16 +296,47 @@ public string FormattedMessage
         /// <summary>
         /// Gets the dictionary of per-event context properties.
         /// </summary>
-        public IDictionary<object, object> Properties
+        public IDictionary<object,object> Properties { get { return this.properties ?? (this.properties = new PropertiesDictionary()); } }
+
+        /// <summary>
+        /// Gets the named parameters extracted from parsing <see cref="Message"/> as MessageTemplate
+        /// </summary>
+        public IEnumerable<MessageTemplateParameter> MessageTemplateParameters
```

Comment:
Could we go again to our custom interface? Even if it's only implementing `IEnumerable<MessageTemplateParameter>`, it's more flexible for the future.

Reply:
Fixed